lib/rack/dev-mark/middleware.rb

```diff
@@ -22,7 +22,8 @@ def call(env)
 
         headers['X-Rack-Dev-Mark-Env'] = CGI.escape Rack::DevMark.env
 
-        if !Rack::DevMark.tmp_disabled && headers['Content-Type'].to_s =~ %r{\btext/html\b}i
+        redirect = 300 <= status.to_i && status.to_i < 400
+        if !redirect && !Rack::DevMark.tmp_disabled && headers['Content-Type'].to_s =~ %r{\btext/html\b}i
```

Comment:
Line is too long. [105/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.